schema_version: 1.2.0
id: CVE-2021-29557
aliases:
- GHSA-xw93-v57j-fcgh
modified: 2021-05-18T19:59Z
published: 2021-05-14T20:15Z
details: TensorFlow is an end-to-end open source platform for machine learning. An
  attacker can cause a denial of service via a FPE runtime error in `tf.raw_ops.SparseMatMul`.
  The division by 0 occurs deep in Eigen code because the `b` tensor is empty. The
  fix will be included in TensorFlow 2.5.0. We will also cherrypick this commit on
  TensorFlow 2.4.2, TensorFlow 2.3.3, TensorFlow 2.2.3 and TensorFlow 2.1.4, as these
  are also affected and still in supported range.
severity:
- type: CVSS_V3
  score: 5.5
- type: CVSS_V2
  score: 2.1
affected:
- package:
    name: tensorflow
    cpe: cpe:2.3:a:google:tensorflow
- ranges:
  - type: GIT
    repo: https://github.com/tensorflow/tensorflow
    events:
    - introduced: '0'
    - fixed: 7f283ff806b2031f407db64c4d3edcda8fb9f9f5
- database_specific:
    CWE:
    - CWE-369
    CVSS:
      Score: 5.5
      Severity: MEDIUM
      Code: CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H
references:
- type: FIX
  url: https://github.com/tensorflow/tensorflow/commit/7f283ff806b2031f407db64c4d3edcda8fb9f9f5
- type: ADVISORY
  url: https://github.com/tensorflow/tensorflow/security/advisories/GHSA-xw93-v57j-fcgh
